### PR TITLE
Hard-light hazard holotape

### PR DESCRIPTION
**Pull request by ghost**
A port from bay, /vg/, or NT, with a twist!

s1 s2
e1 e2

Adds hard-light holotape projectors. These holograms look like your standard 'police tape' or construction tape for crime scenes or hazardous areas (pls new sprites ausops).
Hard-light holotape will harden when a user tries to pass it that does not have the required access. Users with access can manipulate the tape and lift it for people without access should they want to let people in, and users without access can break the tape using just their hands. Its durability can be compared to a grille, and it blocks projectiles so it can be used as impromptu barricading.
Rearranges some lockers a bit, removes the clipboard from detective's locker, adds projector to some job lockers.

secdoor dooreng

lengthsec
lengtheng

### CODE REVIEW

**Comment by TheVekter** [comment]
Definitely needs a new sprite but otherwise fuckin' great.

**Comment by Durkel** [comment]
Do these have a set up time? I fear people would be dropping these down to block taser/laser shots if they're instant or using them as a escape method.

**Comment by paprka** [comment]
Oh, good point. I'll make them so you can only use them if you have the required access.

**Comment by dannno** [comment]
isn't it a bit weiiiirrrd that they block projectiles at all? is that really a good idea?

**Comment by TheVekter** [comment]
I do find it odd that lasers can pass through glass but not these. Probably
because they're made out of hardlight? Do we have any precedent for
hardlight in ss13?

**Comment by paprka** [comment]
It's much, much better than 'you can't pass this tape because you don't have access'

**Comment by TheVekter** [comment]
Fair enough.

**Comment by ike709** [comment]
CSI: Spess is happening

**Comment by AlexCrimson** [comment]
This is awesome. Now we just need someone to rework Detective and we can truly have CSI: In Space.

**Comment by Jordie0608** [comment]
Changelog.

File: code/game/objects/items/hazardtape.dm [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+/obj/item/holotape/attackby(obj/item/weapon/W as obj, mob/user as mob)
+	user.changeNext_move(CLICK_CD_MELEE)
+	add_fingerprint(user)
+	switch(W.damtype)
+		if(STAMINA)
+			return
+		if(BURN)
+			playsound(loc, 'sound/items/welder.ogg', 80, 1)
+		else
```
Comment (Cheridan) [inline code/game/objects/items/hazardtape.dm]: Is the sound stuff necessary? Wouldn't it just play the hitsound of the attacking object anyway?

Comment (paprka) [inline code/game/objects/items/hazardtape.dm]: It actually isn't for some reason, it's a weird bug, I'll redo this in a second

**Comment by Fox-McCloud** [comment]
@paprka 

Decent feature,I reckon; probably won't be utilized much, but it's still nice; I'd ensure that Xenos can break the tape; otherwise, every time there's a Xeno round it'll be "grab the engineering/sec tape and spam it everywhere to stop them!"

Only suggestion I have is to echo Vekter; let projectiles pass through them.

**Comment by Tenebrosity** [comment]
I've wanted this for ages :+1:

**Comment by Tenebrosity** [comment]
I kind of wish you could pass under it instead of having to break it though. It's good that it stops people so they don't lag into hull breaches, but having to destroy it to get by is just going to be a pain for whoever put it up. People are inevitably not going to believe the tape leads to an unfixed breach or an active crime scene, break it to check and get spaced or contaminate a crime scene. The tape will be gone, other people won't have the warning there to stop them doing the same thing, and the person who put the tape up will have to stop what they're doing and fix it each time.

Maybe the tape could you let you pass when walking?

I know this interferes with the access requirement, but letting people lag into hull breaches because they're an engineer but don't necessarily have a space suit isn't great anyway.

**Comment by paprka** [comment]
I would be okay with walking allowing people to pass it and removing the access restrictions entirely, sure.

**Comment by Tenebrosity** [comment]
Great.

**Comment by Miauw62** [comment]
Just do teneb's idea and make sure that they don't block projectiles and this is good :+1:

**Comment by paprka** [comment]
- Renamed file to 'holotape.dm'
- Changed holotape so you can only cross it if you're walking
- Added the ability to project holotape onto windows

**Comment by paprka** [comment]
Lasers now pass through holotape.

**Comment by Cheridan** [comment]
I tested this, and it seems that tape is still access-based and I can't walk through it. Looks like that change didn't get applied somehow.

**Pull request merged**